### PR TITLE
Update Makefile to support darwin/arm64

### PR DESCRIPTION
**Pull request by anqiuy**
This change was introduced to support for Macs using the M1 chip:

- Add darwin/arm64 target to support Apple silicon

### CODE REVIEW

**Comment by jxlwqq** [comment]
arm64 users need it.

**Comment by jxlwqq** [comment]
Excuse me, can you review this PR, when you are free? @sethvargo

File: Makefile [hunk 1]
```diff
@@ -66,6 +66,7 @@ define make-xc-target
 			--workdir="/go/src/${PROJECT}" \
 			"golang:${GOVERSION}" \
 			env \
+				GO111MODULE="off"
```
Comment (sethvargo) [inline Makefile]: Why turn off modules?

**Review by sethvargo — COMMENTED** [review]
I don't have permission to approve this (I don't work for HashiCorp anymore), but this seems fine. I'm not sure about disabling modules though.

**Comment by jxlwqq** [comment]
https://github.com/jxlwqq/http-echo

This image can be used temporarily, which built by `docker buildx`.

**Comment by larstobi** [comment]
I built multi-arch images at https://github.com/larstobi/http-echo/ keeping the same tiny footprint as the original, and with more architectures built by Github Actions:
* linux/amd64
* linux/arm64
* linux/386
* linux/arm/v7
* linux/ppc64le

Published multi-arch images at:

* Github Container Repo: https://github.com/larstobi/http-echo/pkgs/container/http-echo
* Docker Hub: https://hub.docker.com/r/larstobi/http-echo

File: Makefile (hunk 1 above)
Comment (anqiuy) [reply Makefile]: removed this

**Comment by anqiuy** [comment]
@kschoche could you help to review this?

**Review by tgross — APPROVED** [review]
LGTM

**Comment by vinassefranche** [comment]
Hello everybody! Any plan to release the new image with this fix?

**Comment by jackstine** [comment]
Hey, thank you for all your work on this project.
I also was looking for an arm build of the image. I had to make an image myself, but would love to have the support of pulling the image from the repo if possible.

**Comment by tgross** [comment]
Hi @jackstine! As an org we're in the middle of a slow process of revamping a bunch of "common release tooling" for how we ship things like container images. It's been a while since we've shipped a new image of this application, but we'd need to port it over to the new tooling first. I'll see what I can do to get this done, but just fair warning it's the sort of thing I'll be juggling between other projects. 😀

**Comment by antoineco** [comment]
For whoever is interested in a multi-platform version of this image, I built one for myself here: [antoineco/http-echo](https://hub.docker.com/r/antoineco/http-echo)

**Pull request merged**